File: python/8 kyu/368.py
# https://www.codewars.com/kata/57f6ad55cca6e045d2000627/train/python

# Passed

def square_or_square_root(arr):
    result = []
    for num in arr:
        square_root = num ** 0.5
        if int(square_root) == square_root:
            result.append(int(square_root))
        else:
            num_squared = num ** 2
            result.append(num_squared)
            
    return result

output = square_or_square_root([100, 101, 5, 5, 1, 1])
print(output)

File: nodejs/7 kyu/368.js
// https://www.codewars.com/kata/588817db5fb13af14a000020/train/javascript

// Passed

function timedReading(maxLength, text) {
  let cleanseWord = (word) => {
    let letters = word
      .split("")
      .filter((char) => RegExp("[a-z]").test(char.toLowerCase()));
    let cleansedWord = letters.join("");
    return cleansedWord;
  };
  let words = text.split(" ").map((word) => cleanseWord(word));
  let wordsThatWereRead = words.filter(
    (word) => word.length !== 0 && word.length <= maxLength,
  );
  let result = wordsThatWereRead.length;
  return result;
}

const output = timedReading(4, "The Fox asked the stork, 'How is the soup?'");
console.log(output);